components/server/ee/src/user/stripe-service.ts

```diff
@@ -115,15 +116,25 @@ export class StripeService {
         await this.getStripe().subscriptions.del(subscriptionId);
     }
 
-    async createSubscriptionForCustomer(customerId: string): Promise<void> {
+    async createSubscriptionForCustomer(customerId: string, attributionId: string): Promise<void> {
         const customer = await this.getStripe().customers.retrieve(customerId, { expand: ["tax"] });
         if (!customer || customer.deleted) {
             throw new Error(`Stripe customer '${customerId}' could not be found`);
         }
+        const attrId = AttributionId.parse(attributionId);
+        if (!attrId) {
+            throw new Error(`Invalid attributionId '${attributionId}'`);
+        }
         const currency = customer.metadata.preferredCurrency || "USD";
-        const priceId = this.config?.stripeConfig?.usageProductPriceIds[currency];
+        const priceIds =
+            attrId.kind === "team"
```

Comment:
Could we use a switch or if-if else - else here to catch the case when a different `kind` is added such that it doensn't silently fall into the `else` branch?

Reply:
Good point! I agree that such silent fallbacks feel dangerous, so I really appreciate this nudge. 👍 Will fix.